### PR TITLE
support using apt mirrors

### PR DESCRIPTION
**Pull request by paulczar**
Initial support for using apt mirrors for ubuntu and other repos.

### CODE REVIEW

File: roles/haproxy/tasks/main.yml [hunk 1]
```diff
@@ -3,8 +3,11 @@
   template: src=etc/default/haproxy dest=/etc/default/haproxy
   notify: reload haproxy
 
+- name: haproxy ppa key
+  apt_key: id=505d97a41c61b9cd keyserver=keys.gnupg.net state=present
```
Comment (masteinhauser) [inline roles/haproxy/tasks/main.yml]: Will the `keyserver=` still try and call out to that server? How does it handle if it cannot be reached?

**Comment by masteinhauser** [comment]
Tests failed due to:

```
14:29:15 TASK: [ipv6ra | Add bbg/openstack ppa repo] *********************************** 
14:29:15 Tuesday 03 February 2015  14:29:15 -0800 (0:00:00.603)       0:16:25.883 ****** 
14:29:15 fatal: [173.247.105.25] => One or more undefined variables: 'apt' is undefined
14:29:15 fatal: [173.247.105.35] => One or more undefined variables: 'apt' is undefined
```

File: roles/ipv6ra/tasks/main.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   apt_key: id=49DE63CB url='http://keyserver.ubuntu.com:11371/pks/lookup?op=get&search=0xC37BA5F849DE63CB'
```
Comment (masteinhauser) [inline roles/ipv6ra/tasks/main.yml]: Same `keyserver=` question here.

**Comment by masteinhauser** [comment]
A few questions and the failing test, but otherwise looks logical.

**Comment by j2sol** [comment]
This looks good to move on with. Merging.

**Pull request merged**